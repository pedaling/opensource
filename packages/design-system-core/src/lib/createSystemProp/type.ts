import type { Theme } from '../types';

export type InterpolationFn = (props: Record<string, any>) => Record<string, any>;

export type SystemPropConfig = {
  property: string;
  styleProperty?: string;
  scale?: keyof Pick<Theme, 'colors' | 'space' | 'opacity'>;
  disabled?: boolean;
  transform?: (value: any) => Record<string, any> | undefined;
};

export type SystemProp = {
  (props: Record<string, any>): Record<string, any>;
  propName: string;
  disabled: boolean;
};
