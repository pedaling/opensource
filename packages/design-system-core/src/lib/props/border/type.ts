import type { ResponsiveValue } from '../../types';

export type BorderProps = {
  borderWidth?: ResponsiveValue<number>;
  borderStyle?: ResponsiveValue<'solid' | 'none'>;
  borderColor?: ResponsiveValue<string>;
  borderTopWidth?: ResponsiveValue<number>;
  borderTopStyle?: ResponsiveValue<'solid' | 'none'>;
  borderTopColor?: ResponsiveValue<string>;
  borderRightWidth?: ResponsiveValue<number>;
  borderRightStyle?: ResponsiveValue<'solid' | 'none'>;
  borderRightColor?: ResponsiveValue<string>;
  borderBottomWidth?: ResponsiveValue<number>;
  borderBottomStyle?: ResponsiveValue<'solid' | 'none'>;
  borderBottomColor?: ResponsiveValue<string>;
  borderLeftWidth?: ResponsiveValue<number>;
  borderLeftStyle?: ResponsiveValue<'solid' | 'none'>;
  borderLeftColor?: ResponsiveValue<string>;
  borderRadius?: ResponsiveValue<number | string>;
  borderTopLeftRadius?: ResponsiveValue<number | string>;
  borderTopRightRadius?: ResponsiveValue<number | string>;
  borderBottomLeftRadius?: ResponsiveValue<number | string>;
  borderBottomRightRadius?: ResponsiveValue<number | string>;
  outlineWidth?: ResponsiveValue<number>;
  outlineStyle?: ResponsiveValue<'solid' | 'none'>;
  outlineColor?: ResponsiveValue<string>;
  outlineOffset?: ResponsiveValue<number>;
};
