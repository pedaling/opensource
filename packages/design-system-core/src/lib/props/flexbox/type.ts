import type { ResponsiveValue } from '../../types';

export type FlexboxProps = {
  flex?: ResponsiveValue<string | number>;
  flexGrow?: ResponsiveValue<string | number>;
  flexShrink?: ResponsiveValue<string | number>;
  flexBasis?: ResponsiveValue<string | number>;
  flexDirection?: ResponsiveValue<'column' | 'column-reverse' | 'row' | 'row-reverse'>;
  flexWrap?: ResponsiveValue<'nowrap' | 'wrap' | 'wrap-reverse'>;
  alignContent?: ResponsiveValue<
    | 'space-around'
    | 'space-between'
    | 'space-evenly'
    | 'stretch'
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'start'
    | 'baseline'
    | 'normal'
  >;
  alignItems?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'baseline'
    | 'normal'
    | 'stretch'
  >;
  alignSelf?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'auto'
    | 'baseline'
    | 'normal'
    | 'stretch'
  >;
  justifyContent?: ResponsiveValue<
    | 'space-around'
    | 'space-between'
    | 'space-evenly'
    | 'stretch'
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'start'
    | 'left'
    | 'normal'
    | 'right'
  >;
  justifyItems?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'baseline'
    | 'left'
    | 'legacy'
    | 'normal'
    | 'right'
    | 'stretch'
  >;
  justifySelf?: ResponsiveValue<
    | 'center'
    | 'end'
    | 'flex-end'
    | 'flex-start'
    | 'self-end'
    | 'self-start'
    | 'start'
    | 'auto'
    | 'baseline'
    | 'left'
    | 'normal'
    | 'right'
    | 'stretch'
  >;
  order?: ResponsiveValue<number>;
};
