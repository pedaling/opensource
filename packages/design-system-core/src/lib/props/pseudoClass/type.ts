import type { ResponsiveValue } from '@class101/design-system-core';
import type { SystemProps } from '../systemProp';

export type PseudoClassProps = {
  pseudoHover?: ResponsiveValue<SystemProps>;
  pseudoFocus?: ResponsiveValue<SystemProps>;
  pseudoActive?: ResponsiveValue<SystemProps>;
};
