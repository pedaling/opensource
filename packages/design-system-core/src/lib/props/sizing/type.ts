import type { ResponsiveValue } from '../../types';

export type SizingProps = {
  width?: ResponsiveValue<'auto' | 'fit-content' | 'max-content' | 'min-content' | string | number>;
  minWidth?: ResponsiveValue<'auto' | 'fit-content' | 'max-content' | 'min-content' | string | number>;
  maxWidth?: ResponsiveValue<'auto' | 'fit-content' | 'max-content' | 'min-content' | 'none' | string | number>;
  height?: ResponsiveValue<'auto' | 'fit-content' | 'max-content' | 'min-content' | string | number>;
  minHeight?: ResponsiveValue<'auto' | 'fit-content' | 'max-content' | 'min-content' | string | number>;
  maxHeight?: ResponsiveValue<'auto' | 'fit-content' | 'max-content' | 'min-content' | 'none' | string | number>;
};
