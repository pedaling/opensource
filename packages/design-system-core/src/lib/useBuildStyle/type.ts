import type { Theme } from '../types';

type StyleObject = {
  [property: string]: any;
};

export type BuildStyleFn = (styleObjects: StyleObject[], info: { theme?: Theme }) => Record<string, any>;
