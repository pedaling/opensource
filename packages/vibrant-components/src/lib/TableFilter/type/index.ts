export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
