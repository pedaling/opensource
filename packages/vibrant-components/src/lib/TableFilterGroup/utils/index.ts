import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
