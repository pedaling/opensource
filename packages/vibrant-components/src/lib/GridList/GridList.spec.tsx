import type { ReactRenderer } from '@vibrant-ui/utils/testing-web';
import { createReactRenderer } from '@vibrant-ui/utils/testing-web';

describe('<GridList />', () => {
  const { render } = createReactRenderer();

  let renderer: ReactRenderer;

  describe('when columns set', () => {
    // beforeEach(async () => {
    //   renderer = render(<NotificationBadge data-testid="notificationBadge" kind="number" count={5} />);
    //   element = await renderer.getByTestId('notificationBadge');
    // });
    // it('count rendered', () => {
    //   expect(element.children[0].tagName).toBe('SPAN');
    //   expect(element.children[0].textContent).toBe('5');
    // });
    it('the number of columns should be columns', () => {});
  });

  describe('when maxRows set', () => {
    // beforeEach(async () => {
    //   renderer = render(<NotificationBadge data-testid="notificationBadge" kind="number" count={5} />);
    //   element = await renderer.getByTestId('notificationBadge');
    // });
    // it('count rendered', () => {
    //   expect(element.children[0].tagName).toBe('SPAN');
    //   expect(element.children[0].textContent).toBe('5');
    // });
    it('items in a row after maxRow should not be displayed', () => {});
  });

  describe('when onEndReached provided and scrolled to end item', () => {
    it('onEndReached should be called', () => {});
    // beforeEach(async () => {
    //   renderer = render(<NotificationBadge data-testid="notificationBadge" kind="number" count={5} />);
    //   element = await renderer.getByTestId('notificationBadge');
    // });
    // it('count rendered', () => {
    //   expect(element.children[0].tagName).toBe('SPAN');
    //   expect(element.children[0].textContent).toBe('5');
    // });
  });
});
