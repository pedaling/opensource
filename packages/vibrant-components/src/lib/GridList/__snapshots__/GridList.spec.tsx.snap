// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridList /> when the 'breakpoints' prop provided other responsive props should work based on the 'breakpoints' prop, not the theme breakpoints 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  box-sizing: border-box;
  position: relative;
  overflow: auto;
  padding-left: 0;
  padding-right: 0;
  width: 100%;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-column-gap: 0;
  column-gap: 0;
  row-gap: 0;
  scrollbar-width: none;
  -ms-overflow-style: none;
}

.emotion-0::-webkit-scrollbar {
  display: none;
}

.emotion-1 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: calc((100% - 0px) / 1);
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

@media screen and (min-width: 200px) {
  .emotion-1 {
    width: calc((100% - 0px) / 2);
  }
}

@media screen and (min-width: 400px) {
  .emotion-1 {
    width: calc((100% - 0px) / 3);
  }
}

.emotion-2 {
  text-align: left;
  color: #ffffff;
  -webkit-text-fill-color: #ffffff;
  text-fill-color: #ffffff;
  font-size: 1rem;
  line-height: 1.25rem;
  font-weight: 400;
}

<div>
  <ul
    class="emotion-0"
    data-testid="grid-list"
  >
    <li
      class="emotion-1"
    >
      <span
        class="emotion-2"
        data-testid="body"
      >
        1
      </span>
    </li>
    <li
      class="emotion-1"
    >
      <span
        class="emotion-2"
        data-testid="body"
      >
        2
      </span>
    </li>
    <li
      class="emotion-1"
    >
      <span
        class="emotion-2"
        data-testid="body"
      >
        3
      </span>
    </li>
    <li
      class="emotion-1"
    >
      <span
        class="emotion-2"
        data-testid="body"
      >
        4
      </span>
    </li>
    <li
      class="emotion-1"
    >
      <span
        class="emotion-2"
        data-testid="body"
      >
        5
      </span>
    </li>
  </ul>
</div>
`;
